docs/training/example-5.adoc

```diff
@@ -1 +1,400 @@
+
+We have now covered the basic of how to create Beer Garden Plugins with a variety of Input Types and Output Types.
+This Example will merge these concepts together while creating more complex User Forms.
+
+For this example you will need the Beer Statistics Plugin running from the last example.
+
+    python beer_statistics_plugin.py
+
+In the new plugin we are going to create we want to accomplish the following tasks:
+
+. Spawn a child Request that invokes the Beer Statistics Plugin
+. Dynamically create the Drop Down for Year Range
+. Dynamically create the Drop Down for Field Summary Option
+. Create Request that finds the Sum Total for one field from the Beer Statistics Plugin
+
+We are going to work down this list in order for creating our new plugin.
+
+=== Connect to Another Client
+
+In this snippet we are going to create a System Client connection with the Plugin to allow access
+to remote commands.
+
+[source,python]
+.complex_beer_statistics_plugin.py
+----
+
+from brewtils import parameter, command, system, SystemClient, Plugin
+
+@system
+class ComplexBeerStatisticsClient:
+
+    def __init__(self, params):
+        # (1)
+        self.beer_statistics_client = SystemClient(system_name="BeerStatistics", **params)
+
+    @command(output_type="JSON")
+    def test_beer_statistics_client(self):
+        # (2)
+        results = self.beer_statistics_client.download_year_statistics(2010)
+        # (3)
+        return results.output
+
+def main():
+
+    # (4)
+    connection_params = {"bg_host":"<HOST>",
+                         "bg_port":<PORT>,
+                         "ssl_enabled":False}
+
+    client = ComplexBeerStatisticsClient(connection_params)
+
+    plugin = Plugin(
+        client,
+        name='ComplexBeerStatistics',
+        version='1.0.0.dev', # (5)
+        description='Parses data from TTB.gov on Brewer Production sizes of Beer',
+        **connection_params
+    )
+
+    plugin.run()
+
+if __name__ == "__main__":
+    main()
+
+----
+
+<1> Developers can reference other clients through the `SystemClient` class.
+<2> All commands on `SystemClient` can be referenced directly like a standard Class Function
+<3> The output of the `Command` is a `Request`. Developers can put the raw output from the `output` field.
+<4> For ease of use, we are passing forward connection parameters so we only have to define it once
+<5> Because we are going to start and stop this plugin multiple times, using the `.dev` tag on the version
+will not require version increases while you are developing the plugin.
+
+So now lets run this Plugin and see what happens
+
+    python complex_beer_statistics_plugin.py
+
+image::{includedir}/complex-beer-statistics-test.png[Complex Beer Statistics Results]
+
+We can see that a child request was spawned when invoking this command. If you clicked on the hyperlink
+for `download_year_statistics` then you can see the input/output for that command.
+
+=== Building Dynamic Dropdown List
```

Comment:
If you specifically say "Dropdown" you should add that to the `choices` definition, right now they are typeaheads.